spec/factories/transactions.rb

```diff
@@ -4,5 +4,14 @@
     sequence(:credit_card_number) { |n| (1234567890123456 + n).to_s }
     credit_card_expiration_date { "2022-01-03" }
     result { 0 }
+
+    factory :transaction_with_customer do
+      transient do
+        customer {create(:customer)}
+      end
+      after(:create) do |transaction, evaluator|
+        transaction.invoice.update(customer: evaluator.customer)
```

Comment:
Thiiiis is neat! This is calling the created transaction's invoice, then updating that invoices customer information?